### PR TITLE
Make RESTEasy Reactive Client ApplicationScoped by default

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
+ added a note about the client to the resteasy-reactive doc

### CODE REVIEW

File: docs/src/main/asciidoc/resteasy-reactive.adoc [hunk 1]
```diff
@@ -1812,3 +1812,19 @@ public class ResourceForApp1Only {
...
+
+
+== RESTEasy Reactive client
+In addition to the Server side, RESTEasy Reactive comes with a new MicroProfile Rest Client implementation that is non-blocking at its core.
+
+To use it, add the `quarkus-rest-client-microprofile` extension to your project.
+The client can be used as described in https://quarkus.io/guides/rest-client[Using the REST Client guide] and specified in the https://download.eclipse.org/microprofile/microprofile-rest-client-1.4.1/microprofile-rest-client-1.4.1.html[MicroProfile Rest Client specification], with a few exceptions:
+
+- the extension name
+- the default scope of the client for the new extension is `@ApplicationScoped` while the `quarkus-rest-client` defaults to `@Dependent`
```
Comment (geoand) [inline docs/src/main/asciidoc/resteasy-reactive.adoc:1824]: According to `ReactiveResteasyMpClientConfig` the default is `javax.enterprise.context.SessionScoped`.

File: docs/src/main/asciidoc/resteasy-reactive.adoc [hunk 2]
```diff
@@ -1812,3 +1812,19 @@ public class ResourceForApp1Only {
...
+To change this behavior, set the `quarkus.rest.client.scope` property to the fully qualified scope name.
+- it is not possible to set `HostnameVerifier` or `SSLContext`
+- a few things that don't make sense for a non-blocking implementations, such as setting the `ExecutorService`, don't work
```
Comment (geoand) [inline docs/src/main/asciidoc/resteasy-reactive.adoc:1827]: Are these meantt to be somewhere else?

File: docs/src/main/asciidoc/resteasy-reactive.adoc (hunk 1 above)
Comment (michalszynkiewicz) [reply docs/src/main/asciidoc/resteasy-reactive.adoc:1824]: oops, ApplicationScoped is correct, updating. Good catch!

File: docs/src/main/asciidoc/resteasy-reactive.adoc (hunk 2 above)
Comment (michalszynkiewicz) [reply docs/src/main/asciidoc/resteasy-reactive.adoc:1827]: I'm not sure what you mean... Do you mean that the last two limitations seem to be under the info about scope? If yes, that's just  how it looks in adoc, that's how the section looks in the end:

image

Comment (geoand) [reply docs/src/main/asciidoc/resteasy-reactive.adoc:1827]: Ah, OK. I got lost because of the review UI. Looks good

**Review by geoand — APPROVED** [review]

**Pull request merged**